strings/join.py

```diff
@@ -1,51 +1,27 @@
-"""
-Program to join a list of strings with a separator
-"""
-
-
 def join(separator: str, separated: list[str]) -> str:
     """
-    Joins a list of strings using a separator
-    and returns the result.
-
-    :param separator: Separator to be used
-                for joining the strings.
-    :param separated: List of strings to be joined.
-
-    :return: Joined string with the specified separator.
-
-    Examples:
-
-    >>> join("", ["a", "b", "c", "d"])
-    'abcd'
-    >>> join("#", ["a", "b", "c", "d"])
-    'a#b#c#d'
-    >>> join("#", "a")
-    'a'
-    >>> join(" ", ["You", "are", "amazing!"])
-    'You are amazing!'
-
-    This example should raise an
-    exception for non-string elements:
-    >>> join("#", ["a", "b", "c", 1])
-    Traceback (most recent call last):
-        ...
-    Exception: join() accepts only strings
-
-    Additional test case with a different separator:
-    >>> join("-", ["apple", "banana", "cherry"])
-    'apple-banana-cherry'
+    Custom implementation of the join() function.
+    This function manually concatenates the strings in the list,
+    using the provided separator, without relying on str.join().
+
+    :param separator: The separator to place between strings.
+    :param separated: A list of strings to join.
+
+    :return: A single string with elements joined by the separator.
+
+    :raises Exception: If any element in the list is not a string.
     """
+    if not all(isinstance(word_or_phrase, str) for word_or_phrase in separated):
```

Comment:
I think this if statement can be placed inside the for-loop. That way we only iterate over the sequence once, instead of twice.

Reply:
Thank you for your advice. sure lemme check